src/views/List.jsx

```diff
@@ -1,22 +1,49 @@
+import { useState } from 'react';
 import { ListItem } from '../components';
 
 export function List({ data }) {
+	const [listFilter, setListFilter] = useState('');
+
+	const clickHandler = (event) => {
+		event.preventDefault();
+		setListFilter('');
+	};
+
 	return (
 		<>
-			<p>
-				Hello from the <code>/list</code> page!
-			</p>
-			<ul>
-				{data.length > 0 ? (
-					<ul>
-						{data.map((item) => {
+			<form>
+				<label htmlFor="itemInput">Filter Items: </label>
+				<input
+					type="text"
+					id="itemInput"
+					name="itemInput"
+					placeholder="Start typing here..."
+					value={listFilter}
+					onChange={(event) => setListFilter(event.target.value)}
+				/>
+				{listFilter && (
+					<span>
+						<button type="reset" onClick={clickHandler}>
+							X
+						</button>
+					</span>
+				)}
+			</form>
+			{data.length > 0 ? (
+				<ul>
+					{data
+						.filter(
+							(item) =>
+								item.name.toLowerCase().includes(listFilter.toLowerCase()) ||
+								listFilter === '',
```

Comment:
When is this second condition to the logical OR being used?

Reply:
@mentalcaries  hmmm good point. We may have left that in there from something else we were trying and didn't realize it now wasn't necessary. Thanks for catching that! I'll remove it.